Cesium3DTiles/src/SkirtMeshMetadata.cpp

```diff
@@ -47,31 +52,30 @@ SkirtMeshMetadata::parseFromGltfExtras(const JsonValue::Object& extras) {
   }
 
   skirtMeshMetadata.meshCenter = glm::dvec3(
-      (*pMeshCenter)[0].getNumber(0.0),
-      (*pMeshCenter)[1].getNumber(0.0),
-      (*pMeshCenter)[2].getNumber(0.0));
-
-  const double* pWestHeight =
-      gltfSkirtMeshMetadata.getValueForKey<JsonValue::Number>(
-          "skirtWestHeight");
-  const double* pSouthHeight =
-      gltfSkirtMeshMetadata.getValueForKey<JsonValue::Number>(
-          "skirtSouthHeight");
-  const double* pEastHeight =
-      gltfSkirtMeshMetadata.getValueForKey<JsonValue::Number>(
-          "skirtEastHeight");
-  const double* pNorthHeight =
-      gltfSkirtMeshMetadata.getValueForKey<JsonValue::Number>(
-          "skirtNorthHeight");
-
-  if (!pWestHeight || !pSouthHeight || !pEastHeight || !pNorthHeight) {
+      (*pMeshCenter)[0].getSafeNumberOrDefault<double>(0.0),
+      (*pMeshCenter)[1].getSafeNumberOrDefault<double>(0.0),
+      (*pMeshCenter)[2].getSafeNumberOrDefault<double>(0.0));
+
+  double pWestHeight, pSouthHeight, pEastHeight, pNorthHeight;
+  try {
+    pWestHeight = gltfSkirtMeshMetadata.getSafeNumericalValueForKey<double>(
+        "skirtWestHeight");
+    pSouthHeight = gltfSkirtMeshMetadata.getSafeNumericalValueForKey<double>(
+        "skirtSouthHeight");
+    pEastHeight = gltfSkirtMeshMetadata.getSafeNumericalValueForKey<double>(
+        "skirtEastHeight");
+    pNorthHeight = gltfSkirtMeshMetadata.getSafeNumericalValueForKey<double>(
+        "skirtNorthHeight");
+  } catch (const JsonValueMissingKey&) {
```

Comment:
Should we throw in this case if no key is found or value is not correct type? It seems to me returning `nullptr` is a better option, but I may miss something

Reply:
I think I wrote this code to mimic the way it worked before, but I think throwing if the SkritMeshMetadata exists but is malformed isn't a bad idea, that way clients will know immediately their input is malformed. Curious what the majority consensus on this is.

Reply:
I maybe over-concern about the exception in this case. In fact, the exception here probably won't throw as all the fields are double when first created. So it won't cost much. If @jtorresfabra is okay with this, then it's a thumb up for me too

Reply:
I've decided to keep it as is, the current test expects a `std::nullopt` to be returned  if the data is malformed. Keeping it this way unless anyone has strong objections.